package/network/ipv6/6in4/files/6in4.sh

```diff
@@ -44,18 +44,35 @@ proto_6in4_setup() {
 	local cfg="$1"
 	local iface="$2"
 	local link="6in4-$cfg"
+	local remoteip
 
-	local mtu ttl tos ipaddr peeraddr ip6addr ip6prefix ip6prefixes tunlink tunnelid username password updatekey
-	json_get_vars mtu ttl tos ipaddr peeraddr ip6addr tunlink tunnelid username password updatekey
+	local mtu ttl tos ipaddr peeraddr ip6addr ip6prefix ip6prefixes tunlink tunnelid username password updatekey device nohostroute
+	json_get_vars mtu ttl tos ipaddr peeraddr ip6addr tunlink tunnelid username password updatekey device nohostroute
 	json_for_each_item proto_6in4_add_prefix ip6prefix ip6prefixes
 
+	[ -n "$device" ] && link="$device"
+
 	[ -z "$peeraddr" ] && {
-		proto_notify_error "$cfg" "MISSING_ADDRESS"
+		proto_notify_error "$cfg" "MISSING_PEER_ADDRESS"
 		proto_block_restart "$cfg"
 		return
 	}
 
-	( proto_add_host_dependency "$cfg" "$peeraddr" "$tunlink" )
+	remoteip=$(resolveip -t 10 -4 "$peeraddr")
```

Comment:
I wonder whether we can modify this to avoid the new dependency, and use tools already available:
```sh
remoteip=$(nslookup -type=a "$peeraddr" | grep -A 1 "ame" | grep "ddress" | grep -v "#" | awk '{print $2}')
```
I verified that this works on my v23 box.

`resolveip` is perfectly viable, and available in the repos, however.

Reply:
Opted for resolveip because its invocation is short, more readable and already used (and tested) by FQDN endpoint code for OpenWRT GRE and IPIP tunnels.